src/renderer/components/composer/Composer.tsx

```diff
@@ -81,7 +81,28 @@ const Composer = forwardRef<
 
   const chatId = selectedChat.id
   const [showEmojiPicker, setShowEmojiPicker] = useState(false)
+  const [recordingStartTime, setRecordingStartTime] = useState(-1)
+  const [recordedSeconds, setRecordedSeconds] = useState(0)
+  let updateRecordedSecondsInterval: number | 0 = 0
+  
+  let recorder: MediaRecorder | null = null
+  let data: Blob[] = []
+  navigator.mediaDevices.getUserMedia({ audio: true, video: false }).then(stream => {
+    recorder = new MediaRecorder(stream)
+    recorder.onstart = () => {
+    data = []
+    }
+  
+    recorder.ondataavailable = (ev: BlobEvent) => {
+      data.push(ev.data);
+    }
+
+    recorder.onstop = () => {
+      clearInterval(updateRecordedSecondsInterval)
+    }
+  })
```

Comment:
you need to use useEffect, useMemo or useCallback, otherwise this will be run every time the component rerenders